R/epidist_db.R

```diff
@@ -1,22 +1,54 @@
-#' exactly to be recognised.
+#' @description This function can extract an `epidist` object(s) directly from
+#' the library of epidemiological parameters without having to read in an
+#' `epiparam` object and pull out an `epidist` object from one of the entries.
 #'
-#' @details The current criteria for chosing a distribution when multiple
-#' entries exist in the database for a given disease and epidemiological
-#' distribution is to return the entry with the largest sample size.
+#' If a distribution from a specific study is required, the `author` argument
+#' can be specified.
 #'
-#' @param disease A character string specifying the disease
-#' @param epi_dist A character string specifying the epidemiological
-#' distribution
-#' @param author The author of the study reporting the distribution
+#' Multiple entries (`<epidist>` objects) can be returned, use the arguments
+#' to subset entries and use `single_epidist = TRUE` to force a single
+#' `<epidist>` to be returned.
 #'
-#' @return An `epidist` object
+#' @details `disease`, `epi_dist` and `author` are given as individual arguments
+#' as these are the most common variables to subset the parameter library by.
+#' The `subset` argument facilitates all other subsetting of rows to select the
+#' `<epidist>` object(s) desired. To subset based on multiple variables separate
+#' each expression with `&`.
+#'
+#' @param disease A `character` string specifying the disease.
+#' @param epi_dist A `character` string specifying the epidemiological
+#' distribution.
+#' @param author A `character` string specifying the author of the study
+#' reporting the distribution.
+#' @param subset Either `NULL` or a valid R expressions that evaluates to
+#' logicals to subset the rows of `<epiparam>`, or a function that can be
+#' applied directly to an `<epiparam>` object.
+#'
+#' This argument allows general `<data.frame>` subsetting that can be combined
+#' with the subsetting done with the `disease` and `epidist` arguments
+#' (and `author` if specified). If left as `NULL` (default) no subsetting is
+#' carried out.
+#'
+#' The expression should be specified without using the data object name
+#' (e.g. `df$var`) and instead just `var` should be supplied. In
+#' other words, this argument works the same as the `subset` argument in
+#' [`subset()`]. It is similar to `<data-masking>` using by the `dplyr` package.
+#'
+#' @param single_epidist A boolean `logical` determining whether a single
+#' `<epidist>` or multiple entries from the library can be returned if
+#' matched by the other arguments (`disease`, `epi_dist`, `author`). This
+#' argument is used to prevent multiple sets of parameters being returned
+#' when only one is wanted.
+#'
+#' **Note**: If multiple entries match the arguments supplied and
+#' `single_epidist = TRUE` then the `<epidist>` that is parameterised and
+#' has the largest sample size will be returned. If multiple entries
```

Comment:
What happens when:
1. No entries are parameterised?
2. Only one entry is parameterised, but does not have the largest sample size?

Reply:
Okay thanks - worth adding to the docs in a couple of lines? Ignore if already explained in other words